test/unit/eigensolver/test_reduction_to_band.cpp

```diff
@@ -218,19 +218,29 @@ auto allGatherTaus(const SizeType k, const SizeType chunk_size,
 
     const auto this_chunk_size = std::min(k - index_chunk * chunk_size, chunk_size);
 
+    namespace ex = pika::execution::experimental;
+    namespace tt = pika::this_thread::experimental;
+
     std::vector<T> chunk_data;
     if (is_owner) {
       const auto index_chunk_local = to_sizet(index_chunk / comm_grid.size().cols());
       chunk_data = local_taus.at(index_chunk_local);
-      sync::broadcast::send(comm_grid.rowCommunicator(),
-                            common::make_data(chunk_data.data(),
-                                              static_cast<SizeType>(chunk_data.size())));
+
+      tt::sync_wait(ex::schedule(ex::std_thread_scheduler{}) | ex::then([&]() {
+                      sync::broadcast::send(comm_grid.rowCommunicator(),
+                                            common::make_data(chunk_data.data(),
+                                                              static_cast<SizeType>(chunk_data.size())));
```

Comment:
To do: abstract this away somehow or keep explicit? `sync::` functionality does not always need to run on a separate thread (if not on the pika runtime).